### ISSUE TITLE
SDK can freeze after scanning QR code

### ISSUE THREAD

**Issue by kpilyugin** [googlevr/cardboard#182]
While testing iOS native SDK behavior with different headset parameters from 
https://www.hypergridbusiness.com/faq/vr-headset-qr-codes/
we noticed that on some of them app freezes after scanning QR code 
For example, on these ones:
https://www.hypergridbusiness.com/wp-content/uploads/2015/10/Dynamic-Virtual-Viewer-QR-Code-by-reader.png
https://www.hypergridbusiness.com/wp-content/uploads/2018/07/Sharper-Image-qr_viewer_profile-from-reader.png

What would be a good way to validate scanned parameters before creating lens distortion?

**Comment by kafumanto** [comment]
Not sure if it's the same problem, but we have crashes (on both iOS and Android) with the following QR code:

https://i1.wp.com/homido.com/wp-content/uploads/2015/12/homido-1.png?ssl=1

![homido-1](https://user-images.githubusercontent.com/6954066/103224887-f69dfa80-4928-11eb-948a-91991eeee50e.png)

**Comment by jballoffet** [comment]
Thanks for reporting this issue. I confirm I can repro it when scanning those QR codes. About the QR code provided by @kafumanto, the issue is that the serialized protobuf does not contain any left eye FOV angles:
```
{
  "vendor": "VRStream",
  "model": "Homido",
  "screen_to_lens_distance": 0.04399999976158142,
  "inter_lens_distance": 0.05999999865889549,
  "left_eye_field_of_view_angles": [],
  "vertical_alignment": 0,
  "tray_to_lens_distance": 0.03099999949336052,
  "distortion_coefficients": [
    0.20000000298023224,
    0.11999999731779099
  ],
  "primary_button": 0
}
```
what makes [this call](https://github.com/googlevr/cardboard/blob/4382d959a62196405ab75a237b5c39094a137c6e/sdk/device_params/android/device_params.cc#L156) fail and the app crashes. A ticket for supporting this case has been added to our backlog and will be prioritized along with other issues.

About the QR codes provided by @kpilyugin, I don’t see any red flags at first glance:
```
{
  "vendor": "Jay at Play",
  "model": "Dynamic Virtual Viewer",
  "screen_to_lens_distance": 0.05400000140070915,
  "inter_lens_distance": 0.035999998450279236,
  "left_eye_field_of_view_angles": [
    50,
    50,
    50,
    50
  ],
  "vertical_alignment": 0,
  "tray_to_lens_distance": 0.026000000536441803,
  "distortion_coefficients": [
    -0.009999999776482582,
    -0.14000000059604645
  ],
  "primary_button": 0
}
```
But I do see the textures are wrongly distorted, so I suspect there is an issue with the provided distortion coefficients. I have also tried them out using the old Cardboard SDK and the textures are also wrongly distorted. You will find more details about how these coefficients are used in [LensDistortion](https://github.com/googlevr/cardboard/blob/master/sdk/lens_distortion.cc), [DistortionMesh](https://github.com/googlevr/cardboard/blob/master/sdk/distortion_mesh.cc) and [PolynomialRadialDistortion](https://github.com/googlevr/cardboard/blob/master/sdk/polynomial_radial_distortion.cc) classes.

**Comment by chaosemer** [comment]
This was fixed in v1.5.0.
